From akpm@osdl.org Mon Aug 21 15:31:12 2006
Message-Id: <200608212231.k7LMV5Ra022301@shell0.pdx.osdl.net>
From: Eric Sesterhenn <snakebyte@gmx.de>
To: mm-commits@vger.kernel.org
Cc: snakebyte@gmx.de, david-b@pacbell.net, greg@kroah.com
Subject: USB: fix signedness issue in drivers/usb/gadget/ether.c
Date: Mon, 21 Aug 2006 15:31:05 -0700

From: Eric Sesterhenn <snakebyte@gmx.de>

another gcc 4.1 signdness warning:

drivers/usb/gadget/ether.c:2028: warning: comparison of unsigned expression < 0 is always false

length is assigned the value of usb_ep_queue() which returns an int.
Directly after this it is checked for < 0, which can never be true.  Making
length an int makes the error check work again.

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Acked-by: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/ether.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/ether.c
+++ gregkh-2.6/drivers/usb/gadget/ether.c
@@ -2014,7 +2014,7 @@ rndis_control_ack_complete (struct usb_e
 static int rndis_control_ack (struct net_device *net)
 {
 	struct eth_dev          *dev = netdev_priv(net);
-	u32                     length;
+	int                     length;
 	struct usb_request      *resp = dev->stat_req;
 
 	/* in case RNDIS calls this after disconnect */
